src/main/java/org/kiwiproject/dynamicproperties/annotation/DynamicField.java

```diff
@@ -53,10 +57,22 @@
      * Provide a list of possible values for this field by using the given Enum as the list. Useful if field is
      * restricted to a set of values.
      *
-     * @return the choices that should be available for this field base on the given Enum.
+     *  <b>Note:</b> Only one attribute providing choices may be specified per field.
+     *
+     * @return the choices that should be available for this field based on the given Enum.
      */
     Class<? extends Enum> choicesFromEnum() default NullEnum.class;
 
+    /**
+     * Provide a list of possible values with labels and enabled-ness for this field by using the given supplier.
+     * Useful if field is restricted to a set of values.
+     *
+     * <b>Note:</b> Only one attribute providing choices may be specified per field.
```

Comment:
```suggestion
     * <em>Note:</em> Only one attribute providing choices may be specified per field.
```

(The reason is that HTML `b` and `i` tags are deprecated, so we generally prefer to use `strong` and `em` instead.)